woudc_data_registry/models.py

```diff
@@ -279,7 +279,8 @@ class Instrument(base):
 
     id_field = 'instrument_id'
     id_dependencies = ['name', 'model', 'serial',
-                       'station_id', 'dataset_id', 'contributor']
+                       'station_id', 'dataset_id', 'contributor',
+                       'project']
```

Comment:
Still worrying about this line a little bit. The thing is, when you fetch from the SQL instruments table using the registry, it returns an `Instrument` object. It never calls `__init__` on that object, and so the object never has a `contributor` or a `project` attribute. It will only have whatever fields are present in the SQL (i.e. the ones declared at the top of the class). Trying to generate an ID using the non-SQL fields in this list would fail.

I'm not suggesting adding those two fields to the SQL, but perhaps it would be better to have something like this?
`id_dependencies = ['name', 'model', 'serial', 'dataset_id', 'deployment_id']`
That would be equivalent, more or less, since deployment ID contains station ID, contributor, and project. But that way everything would be SQL fields, so this would be safe with `Instrument`s that you fetched straight from the SQL.

You would also have to edit `generate_ids` so that deployment ID is generated first, and only generate it if it's not already there (or alternatively, if both `self.contributor` and `self.project` exist). Hint: look up the `hasattr` builtin function.